test/SlowTests/Server/Documents/DataArchival/DataArchivalDataSubscriptionsTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FastTests;
+using FastTests.Utils;
+using Raven.Client;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Operations.DataArchival;
+using Raven.Client.Documents.Operations.OngoingTasks;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Client.Util;
+using SlowTests.Core.Utils.Entities;
+using Sparrow;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Server.Documents.DataArchival;
+
+
+    public class DataArchivalDataSubscriptionsTests: RavenTestBase
+    {
+        public DataArchivalDataSubscriptionsTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private async Task SetupDataArchival(IDocumentStore store)
+        {
+            var config = new DataArchivalConfiguration {Disabled = false, ArchiveFrequencyInSec = 100};
+
+            await DataArchivalHelper.SetupDataArchival(store, Server.ServerStore, config);
+        }
+        
+        
+    
+        [Fact]
+        public async Task ArchivedDocumentsDataSubscription()
+        {
+            using (var store = GetDocumentStore())
+            {
+                List<Company> companies = new(); 
+                
+                // Insert document with archive time before activating the archival
+                var company = new Company {Name = "Company Name", Address1 = "Dabrowskiego 6"};
+                var retires = SystemTime.UtcNow.AddMinutes(5);
+                using (var session = store.OpenAsyncSession())
+                {
+                    await session.StoreAsync(company);
+                    var metadata = session.Advanced.GetMetadataFor(company);
+                    metadata[Constants.Documents.Metadata.ArchiveAt] = retires.ToString(DefaultFormat.DateTimeOffsetFormatsToWrite);
+                    await session.SaveChangesAsync();
+                }
+                
+                // Set-up the subscription and run the worker
+                var subsId = await store.Subscriptions.CreateAsync(new SubscriptionCreationOptions
+                {
+                    Query = "from Companies",
+                    Name = "Created",
+                    SourceItemKind = IndexSourceItemKind.ArchivedOnly
```

Comment:
We need to have tests with other options as well

Reply:
tests covering that will appear soon